### PR TITLE
feat: add experimental_readRawConfig()

### PR DESCRIPTION
**Pull request by emily-shen**
Fixes  DEVX 1423 

---

- Tests
  - [ ] TODO (before merge)
  - [x] Tests included
  - [ ] Tests not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because: covered by unit tests
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: not user facing change

### CODE REVIEW

File: packages/wrangler/src/config/index.ts [hunk 1]
```diff
@@ -158,498 +152,20 @@ export function readPagesConfig(
 	};
 }
 
-export const readRawConfig = (configPath: string | undefined): RawConfig => {
+export const experimental_readRawConfig = (
```
Comment (edmundhung) [inline packages/wrangler/src/config/index.ts:155]: You need to re-export this on the entry file (`src/cli.ts`) to expose the API

**Review by edmundhung — CHANGES_REQUESTED** [review]

File: packages/wrangler/src/config/index.ts (hunk 1 above)
Comment (emily-shen) [reply packages/wrangler/src/config/index.ts:155]: 🤦 8aecc4ea08ac72924d33d9de91bd331b96b07a46

**Review by penalosa — APPROVED** [review]

File: packages/wrangler/src/__tests__/configuration.test.ts [hunk 2]
```diff
@@ -6037,6 +6038,57 @@ describe("normalizeAndValidateConfig()", () => {
...
+				fs.mkdirSync("../folder", { recursive: true });
+				writeWranglerConfig(
+					{ name: "config-two" },
+					`../folder/wrangler.${configType}`
+				);
+
+				let result = experimental_readRawConfig({
+					script: "./path/to/index.js",
+				});
+				expect(result.rawConfig).toEqual({
```
Comment (vicb) [inline packages/wrangler/src/__tests__/configuration.test.ts:6075]: nit: should this test `result.rawConfig.name` only? (the date is not specified in the config, same on l 6083.

**Review by edmundhung — APPROVED** [review]

File: packages/wrangler/src/__tests__/configuration.test.ts (hunk 2 above)
Comment (emily-shen) [reply packages/wrangler/src/__tests__/configuration.test.ts:6075]: the `writeWranglerConfig` helper provides the default name (`test-name`) and compat date so I think this is okay :)

Comment (emily-shen) [reply packages/wrangler/src/__tests__/configuration.test.ts:6075]: fair enough :) will fix it in a followup PR!

**Pull request merged**